scripts/saveState.js

```diff
@@ -327,9 +330,14 @@ function reloadSandwich () {
 			if (currId.length == 1) {
 				//It's a subgoal
 				//console.log("subgoal");
-				sidebarBody().find("#sideSubgoal" + currId).unbind( "click" ).click(function(){
+			sidebarBody().find("#sideSubgoal" + currId).unbind( "click" ).click(function(){
+				var subArr = getSubgoalArrayFromLocal(); // in case something changes before the button is clicked
+				// do not enter drawSubgoal with a different id until the current subgoal is saved
+				if (statusIsTrue("gotSubgoalQuestions") || currId == subArr.length){
```

Comment:
This should make is so that even on reload/changing the page, actions and subgoals aren't getting overwritten. 🤞

Reply:
Nice!! Is there a subgoal count anywhere to fix the overcount?

Reply:
The subgoal count is incremented in handlePreWalkthroughInfo() in prewalkthrough.js, and I guess this function always runs on load, or just never quite finishes executing :\

Reply:
I don't like how there's duplicate code that essentially does the same thing....i wonder if that's maybe why it sometimes jump to "Subgoal 5" when i'm really on the third subgoal...I'll have to look into this to see what's really going on

Reply:
I can make a new issue! It seems like something that might take a big refactor haha 😅

Reply:
Yes, please! And you're right 😅 this might be harder than it looks, but it'll take some testing to figure it out.